server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,349 @@
+                    f_list.write("{}\n".format(os.path.join(controller, result_name)))
+                    try:
+                        with open(md5) as f:
+                            md5_value = f.readline().split(" ")[0]
+                    except (OSError, IOError) as e:
+                        # Could not open the file
+                        logger.error(
+                            "Could not open the file {}, {}\n".format(md5, e))
+                        continue
+                    md5_returned = md5sum(tar)
+                    if md5_value == md5_returned:
+                        f_ok.write("{}: {}\n".format(os.path.join(controller, result_name), "OK"))
+                    else:
+                        nfailed_md5 += 1
+                        f_fail.write("{}: {}\n".format(os.path.join(controller, result_name), "FAILED"))
+                return nfailed_md5
+            except Exception as e:
+                logger.exception("{}\n".format(e))
+                return sys.maxsize
+    except (OSError, IOError) as e:
+        logger.error(
+            "Could not open one of the temp files for writing {}, {}\n".
+            format((indicator_file, indicator_file_ok, indicator_file_fail), e))
+        return sys.maxsize
+
+
+def report_failed_md5(backup_obj, tmpdir, report, logger):
+    fail_f = os.path.join(tmpdir, "{}.fail", format(backup_obj.list_name))
+    if os.path.exists(fail_f) and os.path.getsize(fail_f) > 0:
+        try:
+            with open(fail_f) as f:
+                failed_list = f.read()
+        except (OSError, IOError) as e:
+            # Could not open the file
+            logger.error(
+                "Could not open the file {}, {}\n".format(fail_f, e))
+        else:
+            report.write(
+                "ERROR: in {}: the calculated MD5 of the following entries "
+                "failed to match the stored MD5:\n {}".format(backup_obj.name, failed_list))
+
+
+def compare_with_s3_backup(s3_config_obj, backup_obj, tmpdir, report, logger):
+
+    if s3_config_obj is None:
+        return Status.FAIL
+
+    # Function to check intergrity of results between local backup and s3
+    list_s3 = os.path.join(tmpdir, "list.s3")
+    try:
+        with open(list_s3, 'w') as f_list:
+            try:
+                s3_content_list = []
+                while True:
+                    resp = s3_config_obj.connector.list_objects(s3_config_obj.bucket_name)
+                    for obj in resp['Contents']:
+                        f_list.write("{}\n".format(obj['Key']))
+                        md5_returned = obj['ETag'].strip("\"")
+                        s3_content_list.append(Entry(obj['Key'], md5_returned))
+                    try:
+                        kwargs['ContinuationToken'] = resp['NextContinuationToken']
+                    except KeyError:
+                        break
+            except Exception as e:
+                logger.exception("{}\n".format(e))
```

Comment:
`logger.exception()` will print out the exception in any case, right? So you don't need to format it specially.  But you should add a meaningful message so we can trace it back to this code.